### PR TITLE
remove core-js

### PR DESCRIPTION
**Pull request by adi518**
This PR removes `core-js` (ES polyfills) and leaves it outside the bundle of this package. Consumers should now add polyfills to their app if they need to support old browsers. I wonder whether it should be released as a new major version, WDYT?

Other stuff this PR does:
- [x] Bumps `@babel/cli` to latest minor (fixes some issues and adds `--no-comments` flag).
- [x] Minify (`--minified` + `--no-comments`) which lowers the bundle size even more.
- [x] Adds source-map generation (let me know if you want to keep it).

Next steps:
Use Webpack 5 or Rollup to build an ESM output, which will make this library treeshakeable, saving some more KBs for consumers. I suggest trying [TSDX](https://tsdx.io/) for this.

PS, Babel CLI docs aren't super updated, so better run `yarn babel -h` to see the latest options.

### CODE REVIEW

File: package.json
Comment (tizmagik) [inline package.json:53]: Do we also need a `NODE_ENV=production` ?

**Comment by tizmagik** [comment]
Thanks, this is great! Do you mind also adding a note in the docs about the needed polyfills for older browser support moving forward after this change?

And yes, to answer your question, I think to be safe we will consider this a major breaking change.

**Comment by tizmagik** [comment]
> Use Webpack 5 or Rollup to build an ESM output, which will make this library treeshakeable, saving some more KBs for consumers. I suggest trying TSDX for this.

Great idea. Happy to accept PRs for these as well (or at least an issue).

File: package.json
Comment (adi518) [reply package.json:53]: I don't think so it, but I'll check.

**Comment by adi518** [comment]
Sure, I'll update the readme.

**Review by tizmagik — APPROVED** [review]
Thanks! I might try and get a minor release out under v8 before merging this for a v9 release but approving and will hold for now 👍

**Comment by patrickskim** [comment]
`core-js` was an issue we've been having as well! Please merge & release ASAP!

**Comment by tizmagik** [comment]
Yes, looking to do a release this week hopefully. I want to land #168 as semver minor in v8 first

**Comment by tizmagik** [comment]
@patrickskim @adi518 looks like #168 might not land this week so I've published this PR against `@alpha` premajor release: 

```
npm install react-tracking@9.0.0-0
```

Try it out and please post back with any issues you come across. Thanks!

**Comment by tizmagik** [comment]
@patrickskim @adi518 have you encountered any issues with this premajor?

**Comment by adi518** [comment]
I haven't used it enough to provide feedback. I soon will.

**Comment by antciccone** [comment]
@tizmagik We were able to upgrade to `react-tracking@9.0.0-0` in our gatsby project. It has been pretty seamless so far. Will update you if @patrickskim or I run into any issues! Thanks for this work 🔥

**Comment by tizmagik** [comment]
[v8.1.0](https://github.com/nytimes/react-tracking/releases/tag/v8.1.0) is released. I'll look to release a v9.0.0 next week or so with this change. Thanks again!

**Comment by adi518** [comment]
We are about two weeks in production with `9.0.0-0` and thus far, no issues.

**Comment by tizmagik** [comment]
RC [v9.0.0-4](https://www.npmjs.com/package/react-tracking/v/9.0.0-4) has been released, mind giving that a quick check and if all looks good I'll merge, tag and release this to mainline @adi518 @antciccone @patrickskim 

```
npm install react-tracking@9.0.0-4
```

**Comment by gedeagas** [comment]
Removing core js make the bundle size much smaller. 15.3kb and 2.42 when gzip. 
Screen Shot 2021-07-15 at 12 41 08 AM

So far no issues, i will run the build on staging for a few days and wait the result. 

Thank you for pushing this @tizmagik 🎉

**Comment by gedeagas** [comment]
@tizmagik after few days of dogfooding and thus far, no issues. I think this release is good to go. 

🚢It 🎉

**Comment by tizmagik** [comment]
Great, thanks @gedeagas !

**Comment by tizmagik** [comment]
Hey @bgergen do you think we can get CI going again? I'd love to have it test this PR before we merge/publish, if possible

**Comment by bgergen** [comment]
@tizmagik Yes! I'll try to get that setup tomorrow. Sorry, I've been swamped today getting ready for Olympics coverage.

**Comment by bgergen** [comment]
@tizmagik Working on figuring out why this isn't triggering a drone build. Maybe I missed something in the config to get it to run on a PR from a fork.

**Comment by tizmagik** [comment]
Thanks @bgergen -- feel free to push directly to this branch

**Comment by tizmagik** [comment]
Yay thank you everyone for all the hard work! 🎉

**Comment by tizmagik** [comment]
This is now [released as v9.0.0](https://github.com/nytimes/react-tracking/releases/tag/v9.0.0) thanks again all!

**Pull request merged**
